commit 84b60e0636e18e2d0ce1cee8eef9a1ab396e40f6
Author: Erik Johansson <erijo@licq.org>
Date:   Sun Oct 17 18:38:17 2010 +0000

    Merged r7908 from trunk as casting function pointers and comparing them doesn't work on hppa so compare the pointers directly

diff --git a/licq/src/utils/tests/dynamiclibrarytest.cpp b/licq/src/utils/tests/dynamiclibrarytest.cpp
index 5e8904b..ef8bd8d 100644
--- licq/src/utils/tests/dynamiclibrarytest.cpp
+++ licq/src/utils/tests/dynamiclibrarytest.cpp
@@ -73,8 +73,8 @@ TEST(DynamicLibrary, loadSelf)
 
   int (*testfuncSymbol)();
   ASSERT_NO_THROW(self.getSymbol("testfunc", &testfuncSymbol));
-  EXPECT_EQ((void*)testfuncSymbol, (void*)&testfunc);
-  EXPECT_EQ(testfuncSymbol(), 42);
+  EXPECT_TRUE(testfunc == testfuncSymbol);
+  EXPECT_EQ(42, testfuncSymbol());
 }
 
 TEST(DynamicLibrary, exceptionGetSystemError)
